data_model/src/domain.rs

```diff
@@ -147,21 +135,12 @@ impl NewDomain {
     }
 }
 
-impl Identifiable for NewDomain {
-    type Id = <Domain as Identifiable>::Id;
-
-    fn id(&self) -> &Self::Id {
-        &self.id
-    }
-}
-
 /// Named group of [`Account`] and [`Asset`](`crate::asset::Asset`) entities.
 #[derive(
     Debug,
     Display,
     Clone,
-    PartialEq,
-    Eq,
+    IdOrdEqHash,
```

Comment:
I assume we settled on this name instead of `Identifiable`?

Reply:
I guess, if nobody minds. It is both succinct and distinct enough to my taste.

Reply:
Yeah, I like it